front/app/modules/commercial/content_builder/admin/components/ContentBuilderMobileView/MobileViewPreview.tsx

```diff
@@ -0,0 +1,109 @@
+import ContentBuilderFrame from '../ContentBuilderFrame';
+import { Box, Spinner, Title } from '@citizenlab/cl2-component-library';
+import { isNilOrError } from 'utils/helperUtils';
+
+// services
+import { PROJECT_DESCRIPTION_CODE } from '../../../services/contentBuilder';
+
+// types
+import { SerializedNodes } from '@craftjs/core';
+
+export const MobileViewPreview = () => {
+  const [draftData, setDraftData] = useState<SerializedNodes | undefined>();
+  const [selectedLocale, setSelectedLocale] = useState<string | undefined>();
+  const { projectId } = useParams() as { projectId: string };
+  const platformLocale = useLocale();
+  const project = useProject({ projectId });
+
+  const contentBuilderLayout = useContentBuilderLayout({
+    projectId,
+    code: PROJECT_DESCRIPTION_CODE,
+  });
+
+  useEffect(() => {
+    const handleMessage = (e: MessageEvent) => {
+      // Make sure there is a root node in the draft data
+      if (e.origin === window.location.origin && e.data.ROOT) {
+        setDraftData(e.data);
+      }
+      if (e.origin === window.location.origin && e.data.selectedLocale) {
+        setSelectedLocale(e.data.selectedLocale);
+      }
+    };
+    window.addEventListener('message', handleMessage);
+    return () => {
+      window.removeEventListener('message', handleMessage);
+    };
+  }, []);
+
+  if (isNilOrError(platformLocale)) {
+    return null;
+  }
+
+  const locale = selectedLocale || platformLocale;
+  const loadingContentBuilderLayout = contentBuilderLayout === undefined;
+
+  const contentBuilderSavedContent =
+    !isNilOrError(contentBuilderLayout) &&
+    contentBuilderLayout.data.attributes.enabled &&
+    contentBuilderLayout.data.attributes.craftjs_jsonmultiloc[locale];
+
+  const contentBuilderContent = draftData || contentBuilderSavedContent;
+
+  const savedEditorData = !isNilOrError(contentBuilderLayout)
+    ? contentBuilderLayout.data.attributes.craftjs_jsonmultiloc[locale]
+    : undefined;
+
+  const editorData = draftData || savedEditorData;
+
+  return !isNilOrError(project) ? (
+    <FocusOn>
+      <Box
+        display="flex"
+        flexDirection="column"
+        w="100%"
+        zIndex="10000"
+        position="fixed"
+        height="100vh"
+        bgColor="#fff"
+        overflowY="auto"
+        data-testid="contentBuilderMobilePreviewContent"
+      >
+        <Box p="20px">
+          <Title color="colorText" variant="h1">
+            {project.attributes.title_multiloc[locale]}
+          </Title>
+          {loadingContentBuilderLayout && <Spinner />}
+          {!loadingContentBuilderLayout && contentBuilderContent && (
+            <Box>
+              <Editor isPreview={true}>
+                <ContentBuilderFrame editorData={editorData} />
+              </Editor>
+            </Box>
+          )}
+        </Box>
+      </Box>
+    </FocusOn>
+  ) : null;
+};
+
+const MobileViewPreviewModal = () => {
+  const modalPortalElement = document.getElementById('modal-portal');
```

Comment:
Non-blocking as it might affect files not related to this. We could use constants for these (`modal-portal`)